src/components/atoms/InputForm/index.tsx

```diff
@@ -1,69 +1,84 @@
-import { ContainerInput , Field } from './style'
-import React, { useState } from 'react'
+import { useField } from "formik";
+import { Label } from "./label";
+import { ContainerInput, Field, StyledError } from "./style";
 
 interface InputFormProps {
-    id: string;
-    name: string;
-    type?: string;
-    placeholder?: string;
-    label: string
-    width: string
-    height?: string
-    as?: string
-    options?: Array<string>
+  id: string;
+  name: string;
+  type?: string;
+  placeholder?: string;
+  label: string;
+  width: string;
+  height?: string;
+  as?: string;
+  options?: Array<string>;
+  minDate?: string;
+  maxDate?: string;
+  callback?: (value: string) => void;
 }
 
 export default function InputForm({
-    id,
-    name,
-    type,
-    placeholder,
-    label,
-    width,
-    height,
-    as,
-    options,
-} : InputFormProps) {
-    const [selectedValue, setSelectedValue] = useState('');
+  id,
+  name,
+  type,
+  placeholder,
+  label,
+  width,
+  height,
+  as,
+  options,
+  minDate,
+  maxDate,
+  callback,
+}: InputFormProps) {
+  const [field] = useField(name);
 
-    const handleSelectChange = (event: any) => {
-        setSelectedValue(event.target.value);
-        console.log(event.target.value);
-    };
+  const handleSelectChange = (event: any) => {
+    if (callback) {
+      callback(event.target.value);
+    }
 
-    return (
-        <ContainerInput>
-            <Label 
-                htmlFor={id} 
-                name={label} 
-            />
-            {as === 'select' ? 
-                <Field 
-                    id={id}
-                    name={name}
-                    width={width}
-                    height={height}
-                    as={as}
-                    onChange={handleSelectChange}
-                    value={selectedValue}
-                >
-                        {options?.map((op, index) => (
-                            <option key={index} value={index > 0 ? op : ""}>
-                                {op}
-                            </option>
-                        ))}
-                </Field>
-            : 
-                <Field
-                    id={id}
-                    name={name}
-                    type={type}
-                    placeholder={placeholder}
-                    width={width}
-                    height={height}
-                />
-            }
-        </ContainerInput>
-    );
-}
\ No newline at end of file
+    field.onChange(event);
+  };
+
+  return (
+    <ContainerInput>
+      <Label htmlFor={id} name={label} />
+      {as === "select" ? (
+        <>
+          <Field
+            id={id}
+            name={name}
+            width={width}
+            height={height}
+            as={as}
+            onChange={handleSelectChange}
+            value={field.value}
+          >
+            {options?.map((op, index) => (
+              <option key={index} value={index > 0 ? op : ""}>
+                {op}
```

Comment:
Ao fazer listas, é interessante utilizar outra opção ao invés de Index. Vou deixar um link que explica com mais detalhes sobre isso:
https://dev.to/shiv1998/why-not-to-use-index-as-key-in-react-lists-practical-example-3e66